### PR TITLE
ci: make server tests run and pass on a clean CI checkout

### PR DESCRIPTION
**Pull request by halcwb**
## Overview

CI ran `dotnet run ServerTests` but executed **zero tests** while reporting success: the `ServerTests` FAKE target invoked `dotnet test --no-restore` with no `Build` dependency, so on a fresh CI checkout nothing was restored or built. Under .NET 10, `dotnet test` on an unbuilt solution exits 0 with zero tests discovered — so CI was green while testing nothing.

Making tests actually run, and improving failure visibility, surfaced several pre-existing failures that only occur in a clean / CI / Windows / slow-runner environment. This PR fixes the root cause and each surfaced failure. All changes are build-infrastructure or test-only — no domain/clinical source is touched.

Final result: **ubuntu ✓ windows ✓ macOS ✓** for `dotnet run ServerTests` (confirmed green twice across the matrix).

## Further information

Six commits, peeling back one failure at a time:

1. **`ci: run server tests on clean CI checkout`** — `Build.fs`: add `Build ==> ServerTests` so the solution is restored and built before `dotnet test`, and fail the target when zero tests run (guards against a silent "ran nothing" regression). Also fixes two clean-env failures the no-op had hidden:
   - **ZIndex.Tests**: the `dotnet test` adapter discovers the `[<Tests>]` value directly and bypasses `Main.fs`, where the synthetic BST fixture setup was forced. Module init also touched raw ZIndex tables before any test, and because .NET caches a failed type initializer for the life of the process, a single early touch poisoned the types for every subsequent test. Fixed by forcing the fixture at the top of `module Tests` (before any nested-module binding) and making the eager `tot` binding tolerant. Runs on the tracked synthetic `BST*.test` data — no proprietary files needed.
   - **ZForm.Tests**: four `DoseRange` tests render strings via Google Sheets unit mappings (need `GENPRES_URL_ID`, unset in CI). Skipped when it is absent, using the same `loadDotEnv` + `getItem` resolution as `Web.genpresUrlId`, so they skip in CI and run locally.

2. **`ci: surface dotnet test output when ServerTests fails`** — the target replaced `dotnet test` stdout with progress dots, so a failing CI run showed no indication of *what* failed (the original `failwithf` was also dead code: the `dotnet` helper's `addOnExited` throws on a non-zero exit inside `Proc.run`, pre-empting the result handler). Build the process directly and dump the captured output on a non-zero exit. This is what made every subsequent failure diagnosable.

3. **`test(agents): deflake property tests on slow CI`** — two FsCheck property tests read a mutable after a fixed `Thread.Sleep`; under load the agent had not drained its queue, so they read stale state (shrinking to minimal input — a timing race). Replaced with a bounded poll-until-condition.

4. **`test(agents): read files with shared access on Windows`** — 14 FileWriterAgent tests wrote via the agent (which holds the file open with `FileShare.ReadWrite`) then read it back with `File.ReadAllLines`, whose default `FileShare.Read` cannot coexist with an open write handle **on Windows**. The read threw and the helpers' `with _ -> [||]` swallowed it into an empty result. Now read through a `FileStream` opened with `FileShare.ReadWrite ||| FileShare.Delete`. (Linux/macOS share modes are advisory, so the plain read happened to work.)

5. **`test(zindex): log instead of silently swallowing tot init failure`** — review follow-up: the `tot` module-init binding caught all exceptions and used 0, which could let the tests comparing against `tot` pass vacuously. Keep the no-crash behaviour but log the exception.

6. **`test(agents): deterministic, non-blocking waits for agent tests`** — more agent tests (basic message passing, ordering, message types, OnError, recoverable error, high throughput, null messages, disposal) used the same fixed-`Async.Sleep`-then-read-mutable pattern. The key subtlety: the wait used inside `testAsync` bodies **must yield the thread** (`do! Async.Sleep`) rather than block it (`Thread.Sleep`). Expecto runs tests in parallel and the MailboxProcessor agents under test run their loops on thread-pool threads; a blocking wait occupies those threads and **starves the very agents being tested**, so on a few-core runner the posted messages are never processed (a first, all-synchronous attempt made this worse — tests timed out at the full 5s). The async variant yields the thread back to the pool; a synchronous `waitUntil` remains only for the two non-async FsCheck property tests. The disposal test additionally waits for the agent to start (so its `finally` is reached) and for that `finally` to run, instead of sleeping.

## Divergence from plan

n/a — CI/test correctness fix; no implementation plan.

## Verification

Reproduced and fixed in a clean `git worktree` (no `.env`, no proprietary cache), mimicking a fresh CI checkout, and confirmed on the CI matrix:

- **Before:** `dotnet test GenPRES.sln --no-restore` → exit 0, 0 tests, ~2s (matches the original CI log).
- **After (CI matrix):** ubuntu, windows, macOS all green — confirmed on two consecutive matrix runs to rule out a lucky pass on the previously-flaky agent tests.
- Agents.Tests: 48/48, stable across repeated local runs and both CI matrix runs.
- ZIndex.Tests passes 24/24 on synthetic data under both `dotnet test` and `dotnet run`; fixture teardown leaves the data dir clean.
- ZForm.Tests: 6 passed, 6 skipped, 0 failed with `GENPRES_URL_ID` unset; the 4 skipped tests run for real when it is set.
- The 10 other test projects that transitively reference `ZIndex.Lib` were confirmed unaffected (they use demo caches, never touch raw tables).

Reviewer repro: check out the branch into a fresh clone / `git worktree` with no `.env`, then run `dotnet run ServerTests`.

## Author checklist

- [x] Follow the process described in CONTRIBUTING.md.
- [x] Make the changes proposed in the linked issue (if applicable): n/a
- [x] Follow the approach documented in the linked implementation plan (if applicable): n/a
- [x] Are no more than 200 lines of changed code, ideally 25-100. (+151 −34 across 4 files; small per-file, single concern — each commit is independently reviewable.)
- [x] Are not more complex than necessary.
- [x] Cannot easily be split in a way that would make reviewing them significantly easier. (The Build.fs fix is what surfaces the test failures; splitting would leave CI red. Commits are scoped for review.)
- [x] Follow the guidelines specified in DEVELOPMENT.md.
- [x] Have been thoroughly tested.
- [x] Don't introduce new security vulnerabilities.
- [x] Follow the AI/LLM Usage Policy — see disclosure below.

### AI/Vibe Coding Disclosure

- [x] Some or all code in this PR is **vibe coded** (substantially AI-generated).

All changes were authored by Claude Code (Opus 4.8) and reviewed/verified by me. The `.fs` test-file edits (ZIndex.Tests, ZForm.Tests, Agents.Tests) were AI-written under an explicit, scoped overrule of the script-only policy for this test-infrastructure fix; no domain/clinical `.fs` source was touched. Verification: full `dotnet run ServerTests` on a clean worktree, targeted per-project runs, and two green CI matrix runs.

- [x] Added appropriate automated tests. (Restores execution of the existing suite and deflakes it; no behaviour change to tested code.)
- [ ] Updated documentation appropriately. (n/a — no doc changes needed.)
- [x] Added comments that highlight important changes and add justification.

### CODE REVIEW

File: tests/Informedica.ZForm.Tests/Tests.fs [hunk 1]
```diff
@@ -146,6 +146,19 @@ module DoseRangeTests =
...
+    // which requires GENPRES_URL_ID. When it is not configured (e.g. CI, where
+    // only demo/cached data is available) these tests cannot run, so skip them.
+    // The DTO round-trip tests below need no sheet data and always run.
+    // Resolution mirrors ZForm.Lib Web.genpresUrlId exactly.
+    let private hasUrlId =
+        Informedica.Utils.Lib.Env.loadDotEnv () |> ignore
+        Informedica.Utils.Lib.Env.getItem "GENPRES_URL_ID" |> Option.isSome
+
+    /// `test` when GENPRES_URL_ID is configured, otherwise a skipped (pending) test.
+    let testUrl name =
```
[the remark this answers is not in the document]
Comment (halcwb) [reply tests/Informedica.ZForm.Tests/Tests.fs:159]: Keeping `testUrl` public on purpose. The repo's F# coding guideline says never mark a *pure* function `private` — pure helpers should stay public as reusable building blocks; `private` is reserved for cases where exposure is actively harmful. `testUrl` is a pure factory (`string -> TestCaseBuilder`), so it stays public. `hasUrlId` above it is `private` because it performs IO (`loadDotEnv`/env read), which is a different case. So the two aren't inconsistent under this convention.

File: tests/Informedica.ZIndex.Tests/Tests.fs [hunk 2]
```diff
@@ -572,7 +584,15 @@ module Tests =
...
-        let tot = GenPresProduct.get [] |> Array.length
+        // Tolerant of a missing/unresolved product cache: module-level bindings
+        // in this file are evaluated eagerly during Tests static init (before any
+        // test runs), so an unguarded data access here would crash the whole
+        // assembly rather than fail a single test.
+        let tot =
+            try
+                GenPresProduct.get [] |> Array.length
+            with _ ->
+                0
```
[the remark this answers is not in the document]
Comment (halcwb) [reply tests/Informedica.ZIndex.Tests/Tests.fs:599]: Addressed in ac21717d: kept the no-crash-during-static-init fallback but now log the exception (`eprintfn`) instead of silently using 0, so a real data-load failure is visible rather than letting the (currently disabled) tests that compare against `tot` pass vacuously.
